### PR TITLE
Add support for Python 3.8

### PR DESCRIPTION
**Pull request by cjwatson**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,10 @@ matrix:
           python: 3.7
           dist: xenial
           sudo: true
+        - os: linux
+          python: 3.8
+          dist: xenial
```
Comment (jugmac00) [inline .travis.yml]: You can leave out `dist: xenial`.

File: .travis.yml [hunk 2]
```diff
@@ -12,6 +12,10 @@ matrix:
...
+          sudo: true
```
Comment (jugmac00) [inline .travis.yml]: The `sudo` keyword had been deprecated quite a while ago - it does nothing any more and should be removed.

File: .travis.yml (hunk 1 above)
Comment (cjwatson) [reply .travis.yml]: Can I remove it from the 3.7 case too?  I was just copying that.

Comment (mgedmin) [reply .travis.yml]: It's fine, I've plans of updating the travis.yml in alignment with the other packages I've updated for https://github.com/zopefoundation/meta/issues/11 some time this week anyway.

File: .travis.yml [hunk 3]
```diff
@@ -31,6 +35,9 @@ matrix:
         - os: osx
           language: generic
           env: TERRYFY_PYTHON='macpython 3.7.0'
+        - os: osx
+          language: generic
+          env: TERRYFY_PYTHON='macpython 3.8.0'
```
Comment (mgedmin) [inline .travis.yml]: Oh, is 3.8 available from terryfy already!  That's nice!

File: CHANGES.rst [hunk 4]
```diff
@@ -10,6 +10,8 @@ Changes
   subclass inheritance. See `PR 144
   <https://github.com/zopefoundation/zope.interface/pull/144>`_.
 
+- Add support for Python 3.8.
+
```
Comment (mgedmin) [inline CHANGES.rst]: I like two blank lines between ReST sections, but we haven't been consistent with this in the past.

File: .travis.yml (hunk 2 above)
Comment (mgedmin) [reply .travis.yml]: Yes, I've been removing both `dist: xenial` and `sudo: true` as I upgrade the travis ymls.

File: .travis.yml (hunk 3 above)
Comment (mgedmin) [reply .travis.yml]: [Apparently not](https://travis-ci.org/zopefoundation/zope.interface/jobs/610282558?utm_medium=notification&utm_source=github_status).

File: .travis.yml (hunk 1 above)
Comment (jugmac00) [reply .travis.yml]: Yes, it is no longer necessary as xenial now is the default environment.
https://blog.travis-ci.com/2019-04-15-xenial-default-build-environment

File: .travis.yml (hunk 3 above)
Comment (mgedmin) [reply .travis.yml]: I'm not a Mac user and know little about these things, but https://github.com/MacPython/terryfy looks nearly abandoned: last commit in 2018, no issues requesting Python 3.8 support, a README that directs users to https://github.com/matthew-brett/multibuild which "gets more development attention".

There's also https://github.com/joerick/cibuildwheel.

Comment (cjwatson) [reply .travis.yml]: OK, I know nothing about Mac builds.  How about I just drop this for now, and somebody with more clue can set it up?

Comment (cjwatson) [reply .travis.yml]: I've done that now.

Comment (mgedmin) [reply .travis.yml]: That was always my approach too: ignore 3.8 on Mac until some Mac user steps up. ;)

**Comment by mgedmin** [comment]
I wouldn't pay too much attention to coveralls numbers, it seems they're not aggregating their data properly.

**Pull request merged**